src/nspawn/nspawn.c

```diff
@@ -3028,6 +3028,14 @@ static int determine_names(void) {
                         return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Failed to determine path, please use -D or -i.");
         }
 
+        /* Find newest version in case directory ends with "auto.d" */
+        char *arg_image_unresolved = arg_image;
+        arg_image = resolve_auto_dir(arg_image);
+        free(arg_image_unresolved);
+        char *arg_directory_unresolved = arg_directory;
+        arg_directory = resolve_auto_dir(arg_directory);
+        free(arg_directory_unresolved);
```

Comment:
oom checks please.

Reply:
also please do so only if --image= or --directory= are specified, and only for the parameter of that switch. please do not do this if we automatically found a machine image with --machine=/-M.

Reply:
OOM checks added. I didn't test machine part but I moved the calls earlier.